vowpalwabbit/sampling.cc

```diff
@@ -0,0 +1,96 @@
+#include "hash.h"
```

Comment:
It seems like we would need to shift this into a header for a header-only version as well.

Are there any changes in the code below from the old library?

Reply:
Ah, it is changed below.